### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by VDubber**
# Description

Basic Account and User Ingestion. 

## Summary

As a team, we identified pain points of integration development while building out the LastPass integration.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [x] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [x] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [x] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [x] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

**Review by adam-in-ict — DISMISSED** [review]

File: src/steps/account/converter.ts [hunk 1]
```diff
@@ -1,25 +1,27 @@
+import { Entities } from '../constants';
 import {
   createIntegrationEntity,
   Entity,
 } from '@jupiterone/integration-sdk-core';
 
-import { Entities } from '../constants';
+function createAccountId(companyId: string): string {
+  return `orca-account-${companyId}`;
+}
```
Comment (zemberdotnet) [inline src/steps/account/converter.ts:9]: `lastpass-account-...`

File: src/steps/users/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export async function fetchUsers({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const client = createAPIClient(instance.config);
+  await client.iterateUsers(async (user: User) => {
+    const userEntity = await jobState.addEntity(createUserEntity(user));
+
+    const account = (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity;
```
Comment (zemberdotnet) [inline src/steps/users/index.ts:23]: We could move this outside the loop and just get the data once.

File: src/steps/account/converter.ts (hunk 1 above)
Comment (VDubber) [reply src/steps/account/converter.ts:9]: Oh the cost of contexting switching. Thanks for catching this!

File: src/types.ts [hunk 3]
```diff
@@ -1,25 +1,37 @@
...
-export type AcmeUser = Opaque<any, 'AcmeUser'>;
-export type AcmeGroup = Opaque<any, 'AcmeGroup'>;
-*/
+export interface User {
+  username: string;
+  fullname: string;
+  mpstrength: string;
+  created: string; // parseTimePropertyValue()
+  last_pw_change: string;
+  last_login: string;
```
Comment (zemberdotnet) [inline src/types.ts:22]: Not necessary change, but I've recently learned that making properties like these have the `Date` type creates a type error when assigning them in `createIntegrationEntity`. It correctly forces the use of `parseTimePropertyValue`. A nice thing to help prevent errors.

File: src/client.ts [hunk 4]
```diff
@@ -16,109 +17,110 @@ export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
...
+      data: body,
+      method: 'POST',
+    };
   }
 
   /**
-   * Iterates each group resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
+   * iterateUsers iterates
```
Comment (zemberdotnet) [inline src/client.ts:87]: Unfinished comment 👾

**Review by zemberdotnet — APPROVED** [review]

**Review by zemberdotnet — APPROVED** [review]

**Pull request merged**